https://github.com/rust-lang/rust/issues/78406

--- rustc-1.47.0-src/library/std/src/os/linux/fs.rs.orig	2020-11-11 13:29:19.986982610 +0000
+++ rustc-1.47.0-src/library/std/src/os/linux/fs.rs	2020-11-11 13:30:13.510161420 +0000
@@ -18,7 +18,7 @@
     /// Unix platforms. The `os::unix::fs::MetadataExt` trait contains the
     /// cross-Unix abstractions contained within the raw stat.
     ///
-    /// [`stat`]: crate::os::linux::raw::stat
+    /// [`stat`]: struct@crate::os::linux::raw::stat
     ///
     /// # Examples
     ///
