--- rustc-1.53.0-src/compiler/rustc_target/src/spec/mod.rs.orig	2021-06-23 14:53:57.471851010 +0000
+++ rustc-1.53.0-src/compiler/rustc_target/src/spec/mod.rs	2021-06-23 14:54:22.405209828 +0000
@@ -818,7 +818,6 @@
     ("armv7r-none-eabihf", armv7r_none_eabihf),
 
     ("x86_64-pc-solaris", x86_64_pc_solaris),
-    ("x86_64-sun-solaris", x86_64_sun_solaris),
     ("sparcv9-sun-solaris", sparcv9_sun_solaris),
 
     ("x86_64-unknown-illumos", x86_64_unknown_illumos),
--- rustc-1.53.0-src/src/tools/build-manifest/src/main.rs.orig	2021-06-23 14:56:07.376321796 +0000
+++ rustc-1.53.0-src/src/tools/build-manifest/src/main.rs	2021-06-23 14:56:17.518074607 +0000
@@ -139,7 +139,6 @@
     "x86_64-linux-android",
     "x86_64-pc-windows-gnu",
     "x86_64-pc-windows-msvc",
-    "x86_64-sun-solaris",
     "x86_64-pc-solaris",
     "x86_64-unknown-freebsd",
     "x86_64-unknown-illumos",
